### PR TITLE
[Major Updates] Add visualization for Backtracking, Add Permutation Notes, Convert jpg to svg

### PR DESCRIPTION
**Pull request by gazcn007**
# Update Summary
- Add VX dependencies and first visualization for Backtracking
image
- Add notes for `Permutation of Elements`
- Convert jpg to background transparent SVG, also changed the color so that the diagrams support both light and dark mode:
image
image

### CODE REVIEW

File: docs/Patterns/Backtracking/backtracking.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    );
+}
+
+function ParentNode({ node }) {
+    const width = 60;
+    const height = 20;
+    const centerX = -width / 2;
+    const centerY = -height / 2;
+
+    return (
```
Comment (JeffZhang02747) [inline docs/Patterns/Backtracking/backtracking.js]: this is a leaf-node and you can write a function for this

File: docs/Patterns/Backtracking/backtracking.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    const height = 20;
+    const centerX = -width / 2;
+    const centerY = -height / 2;
+    const isRoot = node.depth === 0;
+    const isParent = !!node.children;
+
+    if (isRoot) return <RootNode node={node} />;
+    if (isParent) return <ParentNode node={node} />;
+
+    return (
```
Comment (JeffZhang02747) [inline docs/Patterns/Backtracking/backtracking.js]: this is a leaf-node and you can write it in a function

File: docs/Patterns/Backtracking/backtracking.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+const peach = '#fd9b93';
+const pink = '#fe6e9e';
+const blue = '#03c0dc';
+const green = '#26deb0';
+const plum = '#71248e';
+const lightpurple = '#94f4ef';
+const white = '#ffffff';
+const bg = '#2a404e';
+const bg_white = 'white'
```
Comment (JeffZhang02747) [inline docs/Patterns/Backtracking/backtracking.js]: where is this used?

File: docs/Patterns/Backtracking/backtracking.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    ],
+};
+
+function Node({ node }) {
+    const width = 80;
+    const height = 20;
+    const centerX = -width / 2;
+    const centerY = -height / 2;
+    const isRoot = node.depth === 0;
+    const isParent = !!node.children;
```
Comment (JeffZhang02747) [inline docs/Patterns/Backtracking/backtracking.js]: node.height - zero for leaf nodes

node.height > 0; might be better

File: docs/Patterns/Backtracking/backtracking.js (hunk 3 above)
Comment (gazcn007) [reply docs/Patterns/Backtracking/backtracking.js]: fixed

File: docs/Patterns/Backtracking/backtracking.js (hunk 2 above)
Comment (gazcn007) [reply docs/Patterns/Backtracking/backtracking.js]: we will eventually move these VX dependencies to a different npm package; so we will worry about the structures later. right now it's just a proof of concept

**Pull request merged**